postgresql/src/test/java/org/apache/zeppelin/postgresql/PostgreSqlInterpreterTest.java

```diff
@@ -0,0 +1,1667 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zeppelin.postgresql;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.CallableStatement;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.Date;
+import java.sql.NClob;
+import java.sql.PreparedStatement;
+import java.sql.Ref;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.RowId;
+import java.sql.SQLClientInfoException;
+import java.sql.SQLException;
+import java.sql.SQLWarning;
+import java.sql.SQLXML;
+import java.sql.Savepoint;
+import java.sql.Statement;
+import java.sql.Struct;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.Calendar;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Executor;
+
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.postgresql.PostgreSqlInterpreter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * PostgreSQL interpreter unit tests
+ */
+public class PostgreSqlInterpreterTest {
+  @Before
+  public void setUp() throws Exception {
+  }
+
+  @After
+  public void tearDown() throws Exception {
+  }
+
+  @Test
+  public void test() {
+    PostgreSqlInterpreter t = new MockPostgreSqlInterpreter(new Properties());
+    t.open();
+
+    //simple select test
+    InterpreterResult result = t.interpret("select * from t", null);
+    assertEquals(result.type(), InterpreterResult.Type.TABLE);
+
+    //explain test
+    result = t.interpret("explain select * from t", null);
+    assertEquals(result.type(), InterpreterResult.Type.TEXT);
+    t.close();
+  }
+}
+
+class MockPostgreSqlInterpreter extends PostgreSqlInterpreter {
+
+  public MockPostgreSqlInterpreter(Properties property) {
+    super(property);
+  }
+
+  @Override
+  public Connection getJdbcConnection()
+    throws SQLException {
+    return new MockConnection();
+  }
+}
+
+class MockResultSetMetadata implements ResultSetMetaData {
```

Comment:
I don't know what are the project's rules/preferences but this would be much cleaner with the mocks were created by some mocking library, like Mockito.